docs/product/components/modals.html

```diff
@@ -3,31 +3,122 @@
+                    <td>Fires immediately before hiding the modal</td>
+                </tr>
+                <tr>
+                    <th scope="row"><code class="stacks-code">s-modal:hidden</code></th>
+                    <td>Controller element</td>
+                    <td>Fires immediately after hiding the modal</td>
+                </tr>
+            </tbody>
+        </table>
+    </div>
+    {% header h3 | Helpers %}
+    <div class="overflow-x-auto mb32">
+        <table class="wmn4 s-table s-table__bx-simple">
+            <thead>
+                <tr>
+                    <th class="s-table--cell5" scope="col">Function</th>
+                    <th class="s-table--cell2" scope="col">Parameters</th>
+                    <th scope="col">Description</th>
+                </tr>
+            </thead>
+            <tbody class="fs-caption">
+                <tr>
+                    <th scope="row"><code class="stacks-code">Stacks.toggleModal</code></th>
+                    <td>
+                        <code class="stacks-code">element</code>: the element the <code class="stacks-code">data-controller="s-modal"</code> attribute is on <br><br>
+                        <code class="stacks-code">show</code>: whether to force show/hide the modal; toggles the modal if left undefined
+                    </td>
+                    <td>Helper to manually show an s-modal element via external JS</td>
+                </tr>
+            </tbody>
+        </table>
+    </div>
+</section>
+
 <!-- Example base modal content -->
-<aside class="s-modal js-modal-overlay js-modal-close" id="modal-base" tabindex="-1" role="dialog" aria-labelledby="modal-title" aria-describedby="modal-description" aria-hidden="true">
+<aside class="s-modal js-modal-overlay js-modal-close" id="modal-base" tabindex="-1" role="dialog" aria-labelledby="modal-title" aria-describedby="modal-description" aria-hidden="true"
+    data-controller="s-modal" data-target="s-modal.modal">
     <div class="s-modal--dialog js-modal-dialog" role="document">
         <h1 class="s-modal--header" id="modal-title">Example title</h1>
         <p class="s-modal--body" id="modal-description">Nullam ornare lectus vitae lacus sagittis, at sodales leo viverra. Suspendisse nec nulla dignissim elit varius tempus. Cras viverra neque at imperdiet vehicula. Curabitur condimentum id dolor vitae ultrices. Pellentesque scelerisque nunc sit amet leo fringilla bibendum. Etiam feugiat imperdiet mi, eu blandit arcu cursus a. Pellentesque cursus massa id dolor ullamcorper, at condimentum nunc ultrices.</p>
         <div class="grid gs8 gsx s-modal--footer">
             <button class="grid--cell s-btn s-btn__primary js-modal-primary-btn" type="button">Save changes</button>
-            <button class="grid--cell s-btn js-modal-cancel-btn" type="button">Cancel</button>
+            <button class="grid--cell s-btn js-modal-cancel-btn" type="button" data-action="s-modal#toggle">Cancel</button>
```

Comment:
This would be good to remove all the `js-...` classes from this file for clarity.